src/common/table/Table.tsx

```diff
@@ -262,19 +263,17 @@ const Table = <D extends { id: string }>({
     gotoPage(0);
   }, [gotoPage, state.sortBy, state.filters, state.globalFilter]);
 
-  const initialSortByState = initialState?.sortBy?.[0];
-  const currentSortByState = state.sortBy[0];
+  const initialSortByState = initialState?.sortBy;
+  const currentSortByState = state.sortBy;
+  const didSortByChanged = equal(initialSortByState, currentSortByState);
 
   useEffect(() => {
-    if (initialSortByState?.id === currentSortByState?.id && initialSortByState?.desc === currentSortByState?.desc)
-      return;
-
-    onSortedColChange?.(currentSortByState);
-  }, [currentSortByState, onSortedColChange, initialSortByState]);
+    onSortedColChange?.(currentSortByState[0]);
```

Comment:
Should this check if `didSortByChanged === true` before calling `onSortedColChange`?

Reply:
Added